packages/fuchsia_ctl/lib/src/ssh_client.dart

```diff
@@ -92,21 +97,84 @@ class SshClient {
   Future<OperationResult> runCommand(String targetIp,
       {@required String identityFilePath,
       @required List<String> command,
-      Duration timeoutMs = defaultSshTimeoutMs}) async {
+      Duration timeoutMs = defaultSshTimeoutMs,
+      String logFilePath,
+      FileSystem fs}) async {
     assert(targetIp != null);
     assert(identityFilePath != null);
     assert(command != null);
 
-    return OperationResult.fromProcessResult(
-      await processManager
-          .run(
-            getSshArguments(
-              identityFilePath: identityFilePath,
-              targetIp: targetIp,
-              command: command,
-            ),
-          )
-          .timeout(timeoutMs),
+    final bool logToFile = !(logFilePath == null || logFilePath.isEmpty);
+    FileSystem fileSystem = fs ?? const LocalFileSystem();
+    IOSink logFile;
+    Logger logger;
+
+    // If no file is passed to this method we create a memoryfile to keep to
+    // return the stdout in OperationResult.
+    if (logToFile) {
+      fileSystem.file(logFilePath).existsSync() ??
+          fileSystem.file(logFilePath).deleteSync();
+      fileSystem.file(logFilePath).createSync();
+      final IOSink data = fileSystem.file(logFilePath).openWrite();
+      logger = PrintLogger(out: data);
+    } else {
+      fileSystem = MemoryFileSystem();
+      fileSystem.file('logs')..createSync();
+      logFile = fileSystem.file('logs').openWrite();
```

Comment:
This looks like it's not used below and the output will just be an empty string if no log file is specified.

I think we might be better off having some kind of `BufferedLogger` class that would allow us to retrieve the buffer as well as writing it to a file if needed.

Alternatively, if we're logging this to a file/stdout, do we really need to return stdout/stderr back to the caller?

Reply:
Removed from here and moved the functionality to the logger class.